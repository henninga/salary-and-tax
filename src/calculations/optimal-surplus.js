import calculations from './index';


export const optimalSurplus = (parameters) => {
  const initialCalculation = calculations(parameters);
  const companyIncome = initialCalculation.salary.companyIncome;
  const step = 10000;
  const steps = Math.round(companyIncome / step);

  return Array.from(Array(steps).keys())
    .map(i => i * step)
    .map(surplus => calculations({...parameters, surplus: {amount: surplus, include: true}}))
    .map(({afterTotal, dividends: {afterDividendsTax, surplus}, salary: {withoutPension}}) => ({
      total: afterTotal, surplus, dividends: afterDividendsTax, salary: withoutPension,
    }))
};

export default optimalSurplus;
