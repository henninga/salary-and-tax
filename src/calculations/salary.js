import { translationConfig } from '../translation';
import { getPension, pensionConstants } from './pension';
import round from './round';


export const defaultSalaryParameters = {
  cut: 10,
  hoursPerYear: 1730,
  hourRate: 1100,
  includeVacationSavings: true,
  vacationSavingsRate: 12,
  includeEmployerFee: true,
  employerFeeRate: 14.1,
  ensuringFee: 15000,
  locale: translationConfig.locale, ...pensionConstants,
};

export const getTurnover = ({hoursPerYear, hourRate}) => ({turnover: parseFloat(hoursPerYear) * parseFloat(hourRate)});
export const getCompanyIncome = (turnover, {cut}) => {
  const theCut = turnover * parseFloat(cut) / 100;
  return ({
    theCut: theCut, after: turnover - theCut,
  });
};
export const getEmployerFee = (salary, {includeEmployerFee, employerFeeRate}) => {
  const employerFee = round(salary - (salary / (100 + (includeEmployerFee ? parseFloat(employerFeeRate) : 0)) * 100));
  return {
    employerFee, after: salary - employerFee,
  };
};
export const getVacationSavings = (salary, {includeVacationSavings, vacationSavingsRate}) => {
  const vacationSavings = round(salary - (salary / (100 + (includeVacationSavings ? parseFloat(vacationSavingsRate) : 0)) * 100));
  return {
    vacationSavings, after: salary - vacationSavings,
  };
};

export const getSalaryCalculations = (data) => {
  const {turnover} = getTurnover(data);
  const {theCut, after: companyIncome} = getCompanyIncome(turnover, data);
  const {employerFee, after: withoutEmployerFee} = getEmployerFee(companyIncome, data);
  const {vacationSavings, after: withoutVacationSavings} = getVacationSavings(withoutEmployerFee, data);
  const {pension, after: withoutPension} = getPension(withoutVacationSavings, data);
  const totalCuts = theCut + employerFee + vacationSavings + pension;

  return ({
    turnover,
    theCut,
    companyIncome,
    employerFee,
    withoutEmployerFee,
    vacationSavings,
    withoutVacationSavings,
    pension,
    withoutPension,
    income: withoutPension,
    totalCuts,
  });
};
