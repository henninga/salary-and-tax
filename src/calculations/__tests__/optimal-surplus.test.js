import { optimalSurplus } from '../optimal-surplus';
import { defaultSalaryParameters } from '../salary';


describe('optimal-surplus', () => {
  it('should find max', function () {
    expect(optimalSurplus(defaultSalaryParameters)).toMatchSnapshot();
  });
});
