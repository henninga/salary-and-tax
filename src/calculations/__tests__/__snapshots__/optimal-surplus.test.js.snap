// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optimal-surplus should find max 1`] = `
Array [
  Object {
    "dividends": 0,
    "salary": 1318912,
    "surplus": 0,
    "total": 816919,
  },
  Object {
    "dividends": 5344.57,
    "salary": 1311087,
    "surplus": 10000,
    "total": 818084.57,
  },
  Object {
    "dividends": 10689.14,
    "salary": 1303262,
    "surplus": 20000,
    "total": 819251.14,
  },
  Object {
    "dividends": 16033.71,
    "salary": 1295437,
    "surplus": 30000,
    "total": 820416.71,
  },
  Object {
    "dividends": 21378.28,
    "salary": 1287612,
    "surplus": 40000,
    "total": 821583.28,
  },
  Object {
    "dividends": 26722.85,
    "salary": 1279787,
    "surplus": 50000,
    "total": 822748.85,
  },
  Object {
    "dividends": 32067.42,
    "salary": 1271962,
    "surplus": 60000,
    "total": 823915.42,
  },
  Object {
    "dividends": 37411.990000000005,
    "salary": 1264137,
    "surplus": 70000,
    "total": 825081.99,
  },
  Object {
    "dividends": 42756.56,
    "salary": 1256311,
    "surplus": 80000,
    "total": 826246.56,
  },
  Object {
    "dividends": 48101.130000000005,
    "salary": 1248486,
    "surplus": 90000,
    "total": 827413.13,
  },
  Object {
    "dividends": 53445.7,
    "salary": 1240661,
    "surplus": 100000,
    "total": 828578.7,
  },
  Object {
    "dividends": 58790.270000000004,
    "salary": 1232836,
    "surplus": 110000,
    "total": 829744.27,
  },
  Object {
    "dividends": 64134.84,
    "salary": 1225010,
    "surplus": 120000,
    "total": 830909.84,
  },
  Object {
    "dividends": 69479.41,
    "salary": 1217185,
    "surplus": 130000,
    "total": 832076.41,
  },
  Object {
    "dividends": 74823.98000000001,
    "salary": 1209360,
    "surplus": 140000,
    "total": 833240.98,
  },
  Object {
    "dividends": 80168.55,
    "salary": 1201535,
    "surplus": 150000,
    "total": 834407.55,
  },
  Object {
    "dividends": 85513.12,
    "salary": 1193709,
    "surplus": 160000,
    "total": 835574.12,
  },
  Object {
    "dividends": 90857.69,
    "salary": 1185884,
    "surplus": 170000,
    "total": 836740.69,
  },
  Object {
    "dividends": 96202.26000000001,
    "salary": 1178059,
    "surplus": 180000,
    "total": 837906.26,
  },
  Object {
    "dividends": 101546.83,
    "salary": 1170234,
    "surplus": 190000,
    "total": 839071.83,
  },
  Object {
    "dividends": 106891.4,
    "salary": 1162408,
    "surplus": 200000,
    "total": 840237.4,
  },
  Object {
    "dividends": 112235.97,
    "salary": 1154583,
    "surplus": 210000,
    "total": 841402.97,
  },
  Object {
    "dividends": 117580.54000000001,
    "salary": 1146758,
    "surplus": 220000,
    "total": 842569.54,
  },
  Object {
    "dividends": 122925.11,
    "salary": 1138980,
    "surplus": 230000,
    "total": 843761.11,
  },
  Object {
    "dividends": 128269.68,
    "salary": 1131310,
    "surplus": 240000,
    "total": 845009.6799999999,
  },
  Object {
    "dividends": 133614.25,
    "salary": 1123642,
    "surplus": 250000,
    "total": 846259.25,
  },
  Object {
    "dividends": 138958.82,
    "salary": 1115973,
    "surplus": 260000,
    "total": 847508.8200000001,
  },
  Object {
    "dividends": 144303.39,
    "salary": 1108305,
    "surplus": 270000,
    "total": 848758.39,
  },
  Object {
    "dividends": 149647.96000000002,
    "salary": 1100635,
    "surplus": 280000,
    "total": 850006.96,
  },
  Object {
    "dividends": 154992.53,
    "salary": 1092967,
    "surplus": 290000,
    "total": 851257.53,
  },
  Object {
    "dividends": 160337.1,
    "salary": 1085298,
    "surplus": 300000,
    "total": 852507.1,
  },
  Object {
    "dividends": 165681.66999999998,
    "salary": 1077630,
    "surplus": 310000,
    "total": 853755.6699999999,
  },
  Object {
    "dividends": 171026.24,
    "salary": 1069961,
    "surplus": 320000,
    "total": 855005.24,
  },
  Object {
    "dividends": 176370.81,
    "salary": 1062292,
    "surplus": 330000,
    "total": 856254.81,
  },
  Object {
    "dividends": 181715.38,
    "salary": 1054624,
    "surplus": 340000,
    "total": 857505.38,
  },
  Object {
    "dividends": 187059.95,
    "salary": 1046955,
    "surplus": 350000,
    "total": 858754.95,
  },
  Object {
    "dividends": 192404.52000000002,
    "salary": 1039287,
    "surplus": 360000,
    "total": 860003.52,
  },
  Object {
    "dividends": 197749.09,
    "salary": 1031617,
    "surplus": 370000,
    "total": 861252.09,
  },
  Object {
    "dividends": 203093.66,
    "salary": 1023949,
    "surplus": 380000,
    "total": 862502.66,
  },
  Object {
    "dividends": 208438.22999999998,
    "salary": 1016280,
    "surplus": 390000,
    "total": 863752.23,
  },
  Object {
    "dividends": 213782.8,
    "salary": 1008612,
    "surplus": 400000,
    "total": 865001.8,
  },
  Object {
    "dividends": 219127.37,
    "salary": 1000943,
    "surplus": 410000,
    "total": 866250.37,
  },
  Object {
    "dividends": 224471.94,
    "salary": 993275,
    "surplus": 420000,
    "total": 867499.94,
  },
  Object {
    "dividends": 229816.51,
    "salary": 985606,
    "surplus": 430000,
    "total": 868749.51,
  },
  Object {
    "dividends": 235161.08000000002,
    "salary": 977938,
    "surplus": 440000,
    "total": 870000.0800000001,
  },
  Object {
    "dividends": 240505.65,
    "salary": 970268,
    "surplus": 450000,
    "total": 871248.65,
  },
  Object {
    "dividends": 245850.22,
    "salary": 962600,
    "surplus": 460000,
    "total": 872498.22,
  },
  Object {
    "dividends": 251194.78999999998,
    "salary": 954931,
    "surplus": 470000,
    "total": 873534.79,
  },
  Object {
    "dividends": 256539.36,
    "salary": 947263,
    "surplus": 480000,
    "total": 874554.36,
  },
  Object {
    "dividends": 261883.93,
    "salary": 939593,
    "surplus": 490000,
    "total": 875572.9299999999,
  },
  Object {
    "dividends": 267228.5,
    "salary": 931925,
    "surplus": 500000,
    "total": 876592.5,
  },
  Object {
    "dividends": 272573.07,
    "salary": 924256,
    "surplus": 510000,
    "total": 877611.0700000001,
  },
  Object {
    "dividends": 277917.64,
    "salary": 916588,
    "surplus": 520000,
    "total": 878631.64,
  },
  Object {
    "dividends": 283262.21,
    "salary": 908918,
    "surplus": 530000,
    "total": 879650.21,
  },
  Object {
    "dividends": 288606.78,
    "salary": 901250,
    "surplus": 540000,
    "total": 880670.78,
  },
  Object {
    "dividends": 293951.35,
    "salary": 893581,
    "surplus": 550000,
    "total": 881689.35,
  },
  Object {
    "dividends": 299295.92000000004,
    "salary": 885913,
    "surplus": 560000,
    "total": 882708.92,
  },
  Object {
    "dividends": 304640.49,
    "salary": 878243,
    "surplus": 570000,
    "total": 883727.49,
  },
  Object {
    "dividends": 309985.06,
    "salary": 870575,
    "surplus": 580000,
    "total": 884748.06,
  },
  Object {
    "dividends": 315329.63,
    "salary": 862906,
    "surplus": 590000,
    "total": 885767.63,
  },
  Object {
    "dividends": 320674.2,
    "salary": 855238,
    "surplus": 600000,
    "total": 886787.2,
  },
  Object {
    "dividends": 326018.77,
    "salary": 847568,
    "surplus": 610000,
    "total": 887805.77,
  },
  Object {
    "dividends": 331363.33999999997,
    "salary": 839900,
    "surplus": 620000,
    "total": 888825.34,
  },
  Object {
    "dividends": 336707.91000000003,
    "salary": 832231,
    "surplus": 630000,
    "total": 889844.91,
  },
  Object {
    "dividends": 342052.48,
    "salary": 824563,
    "surplus": 640000,
    "total": 890864.48,
  },
  Object {
    "dividends": 347397.05,
    "salary": 816893,
    "surplus": 650000,
    "total": 891883.05,
  },
  Object {
    "dividends": 352741.62,
    "salary": 809225,
    "surplus": 660000,
    "total": 892903.62,
  },
  Object {
    "dividends": 358086.19,
    "salary": 801556,
    "surplus": 670000,
    "total": 893921.19,
  },
  Object {
    "dividends": 363430.76,
    "salary": 793888,
    "surplus": 680000,
    "total": 894941.76,
  },
  Object {
    "dividends": 368775.32999999996,
    "salary": 786219,
    "surplus": 690000,
    "total": 895961.33,
  },
  Object {
    "dividends": 374119.9,
    "salary": 778550,
    "surplus": 700000,
    "total": 896980.9,
  },
  Object {
    "dividends": 379464.47,
    "salary": 770881,
    "surplus": 710000,
    "total": 898000.47,
  },
  Object {
    "dividends": 384809.04000000004,
    "salary": 763213,
    "surplus": 720000,
    "total": 899020.04,
  },
  Object {
    "dividends": 390153.61,
    "salary": 755544,
    "surplus": 730000,
    "total": 900038.61,
  },
  Object {
    "dividends": 395498.18,
    "salary": 747875,
    "surplus": 740000,
    "total": 901058.1799999999,
  },
  Object {
    "dividends": 400842.75,
    "salary": 740206,
    "surplus": 750000,
    "total": 902077.75,
  },
  Object {
    "dividends": 406187.32,
    "salary": 732538,
    "surplus": 760000,
    "total": 903097.3200000001,
  },
  Object {
    "dividends": 411531.89,
    "salary": 724869,
    "surplus": 770000,
    "total": 904115.89,
  },
  Object {
    "dividends": 416876.45999999996,
    "salary": 717201,
    "surplus": 780000,
    "total": 905136.46,
  },
  Object {
    "dividends": 422221.03,
    "salary": 709532,
    "surplus": 790000,
    "total": 906155.03,
  },
  Object {
    "dividends": 427565.6,
    "salary": 701864,
    "surplus": 800000,
    "total": 907175.6,
  },
  Object {
    "dividends": 432910.17000000004,
    "salary": 694195,
    "surplus": 810000,
    "total": 908194.17,
  },
  Object {
    "dividends": 438254.74,
    "salary": 686526,
    "surplus": 820000,
    "total": 909213.74,
  },
  Object {
    "dividends": 443599.31,
    "salary": 678857,
    "surplus": 830000,
    "total": 910233.31,
  },
  Object {
    "dividends": 448943.88,
    "salary": 671188,
    "surplus": 840000,
    "total": 911252.88,
  },
  Object {
    "dividends": 454288.45,
    "salary": 663519,
    "surplus": 850000,
    "total": 912271.45,
  },
  Object {
    "dividends": 459633.02,
    "salary": 655850,
    "surplus": 860000,
    "total": 913291.02,
  },
  Object {
    "dividends": 464977.58999999997,
    "salary": 648182,
    "surplus": 870000,
    "total": 914310.59,
  },
  Object {
    "dividends": 470322.16000000003,
    "salary": 640513,
    "surplus": 880000,
    "total": 915330.16,
  },
  Object {
    "dividends": 475666.73,
    "salary": 632845,
    "surplus": 890000,
    "total": 916349.73,
  },
  Object {
    "dividends": 481011.3,
    "salary": 625175,
    "surplus": 900000,
    "total": 917368.3,
  },
  Object {
    "dividends": 486355.87,
    "salary": 617507,
    "surplus": 910000,
    "total": 918387.87,
  },
  Object {
    "dividends": 491700.44,
    "salary": 609838,
    "surplus": 920000,
    "total": 919407.44,
  },
  Object {
    "dividends": 497045.01,
    "salary": 602170,
    "surplus": 930000,
    "total": 920427.01,
  },
  Object {
    "dividends": 502389.57999999996,
    "salary": 594500,
    "surplus": 940000,
    "total": 921122.58,
  },
  Object {
    "dividends": 507734.15,
    "salary": 586832,
    "surplus": 950000,
    "total": 921445.15,
  },
  Object {
    "dividends": 513078.72,
    "salary": 579163,
    "surplus": 960000,
    "total": 921766.72,
  },
  Object {
    "dividends": 518423.29000000004,
    "salary": 571495,
    "surplus": 970000,
    "total": 922087.29,
  },
  Object {
    "dividends": 523767.86,
    "salary": 563826,
    "surplus": 980000,
    "total": 922408.86,
  },
  Object {
    "dividends": 529112.4299999999,
    "salary": 556157,
    "surplus": 990000,
    "total": 922731.4299999999,
  },
  Object {
    "dividends": 534457,
    "salary": 548488,
    "surplus": 1000000,
    "total": 923053,
  },
  Object {
    "dividends": 539801.5700000001,
    "salary": 540820,
    "surplus": 1010000,
    "total": 923375.5700000001,
  },
  Object {
    "dividends": 545146.14,
    "salary": 533151,
    "surplus": 1020000,
    "total": 923697.14,
  },
  Object {
    "dividends": 550490.71,
    "salary": 525482,
    "surplus": 1030000,
    "total": 924016.71,
  },
  Object {
    "dividends": 555835.28,
    "salary": 517813,
    "surplus": 1040000,
    "total": 924338.28,
  },
  Object {
    "dividends": 561179.85,
    "salary": 510145,
    "surplus": 1050000,
    "total": 924660.85,
  },
  Object {
    "dividends": 566524.42,
    "salary": 502476,
    "surplus": 1060000,
    "total": 924982.42,
  },
  Object {
    "dividends": 571868.99,
    "salary": 494807,
    "surplus": 1070000,
    "total": 925303.99,
  },
  Object {
    "dividends": 577213.56,
    "salary": 487138,
    "surplus": 1080000,
    "total": 925625.56,
  },
  Object {
    "dividends": 582558.13,
    "salary": 479470,
    "surplus": 1090000,
    "total": 925946.13,
  },
  Object {
    "dividends": 587902.7,
    "salary": 471801,
    "surplus": 1100000,
    "total": 926267.7,
  },
  Object {
    "dividends": 593247.27,
    "salary": 464132,
    "surplus": 1110000,
    "total": 926589.27,
  },
  Object {
    "dividends": 598591.8400000001,
    "salary": 456463,
    "surplus": 1120000,
    "total": 926910.8400000001,
  },
  Object {
    "dividends": 603936.4099999999,
    "salary": 448795,
    "surplus": 1130000,
    "total": 927232.4099999999,
  },
  Object {
    "dividends": 609280.98,
    "salary": 441126,
    "surplus": 1140000,
    "total": 927554.98,
  },
  Object {
    "dividends": 614625.55,
    "salary": 433457,
    "surplus": 1150000,
    "total": 927876.55,
  },
  Object {
    "dividends": 619970.12,
    "salary": 425788,
    "surplus": 1160000,
    "total": 928197.12,
  },
  Object {
    "dividends": 625314.69,
    "salary": 418120,
    "surplus": 1170000,
    "total": 928519.69,
  },
  Object {
    "dividends": 630659.26,
    "salary": 410451,
    "surplus": 1180000,
    "total": 928841.26,
  },
  Object {
    "dividends": 636003.8300000001,
    "salary": 402783,
    "surplus": 1190000,
    "total": 929162.8300000001,
  },
  Object {
    "dividends": 641348.4,
    "salary": 395114,
    "surplus": 1200000,
    "total": 929484.4,
  },
  Object {
    "dividends": 646692.97,
    "salary": 387446,
    "surplus": 1210000,
    "total": 929805.97,
  },
  Object {
    "dividends": 652037.54,
    "salary": 379777,
    "surplus": 1220000,
    "total": 930127.54,
  },
  Object {
    "dividends": 657382.11,
    "salary": 372108,
    "surplus": 1230000,
    "total": 930448.11,
  },
  Object {
    "dividends": 662726.6799999999,
    "salary": 364439,
    "surplus": 1240000,
    "total": 930769.6799999999,
  },
  Object {
    "dividends": 668071.25,
    "salary": 356771,
    "surplus": 1250000,
    "total": 931092.25,
  },
  Object {
    "dividends": 673415.8200000001,
    "salary": 349102,
    "surplus": 1260000,
    "total": 931413.8200000001,
  },
  Object {
    "dividends": 678760.39,
    "salary": 341433,
    "surplus": 1270000,
    "total": 931734.39,
  },
  Object {
    "dividends": 684104.96,
    "salary": 333764,
    "surplus": 1280000,
    "total": 932055.96,
  },
  Object {
    "dividends": 689449.53,
    "salary": 326096,
    "surplus": 1290000,
    "total": 932378.53,
  },
  Object {
    "dividends": 694794.1,
    "salary": 318427,
    "surplus": 1300000,
    "total": 932700.1,
  },
  Object {
    "dividends": 700138.6699999999,
    "salary": 310759,
    "surplus": 1310000,
    "total": 933022.6699999999,
  },
  Object {
    "dividends": 705483.24,
    "salary": 303089,
    "surplus": 1320000,
    "total": 933343.24,
  },
  Object {
    "dividends": 710827.81,
    "salary": 295421,
    "surplus": 1330000,
    "total": 933663.81,
  },
  Object {
    "dividends": 716172.38,
    "salary": 287752,
    "surplus": 1340000,
    "total": 933985.38,
  },
  Object {
    "dividends": 721516.95,
    "salary": 280084,
    "surplus": 1350000,
    "total": 934307.95,
  },
  Object {
    "dividends": 726861.52,
    "salary": 272414,
    "surplus": 1360000,
    "total": 934628.52,
  },
  Object {
    "dividends": 732206.0900000001,
    "salary": 264746,
    "surplus": 1370000,
    "total": 934951.0900000001,
  },
  Object {
    "dividends": 737550.6599999999,
    "salary": 257077,
    "surplus": 1380000,
    "total": 935272.6599999999,
  },
  Object {
    "dividends": 742895.23,
    "salary": 249409,
    "surplus": 1390000,
    "total": 935593.23,
  },
  Object {
    "dividends": 748239.8,
    "salary": 241739,
    "surplus": 1400000,
    "total": 935913.8,
  },
  Object {
    "dividends": 753584.37,
    "salary": 234071,
    "surplus": 1410000,
    "total": 936164.37,
  },
  Object {
    "dividends": 758928.94,
    "salary": 226402,
    "surplus": 1420000,
    "total": 936339.94,
  },
  Object {
    "dividends": 764273.51,
    "salary": 218734,
    "surplus": 1430000,
    "total": 936516.51,
  },
  Object {
    "dividends": 769618.0800000001,
    "salary": 211064,
    "surplus": 1440000,
    "total": 936086.0800000001,
  },
  Object {
    "dividends": 774962.65,
    "salary": 203396,
    "surplus": 1450000,
    "total": 935468.65,
  },
  Object {
    "dividends": 780307.22,
    "salary": 195727,
    "surplus": 1460000,
    "total": 934850.22,
  },
  Object {
    "dividends": 785651.79,
    "salary": 188059,
    "surplus": 1470000,
    "total": 934233.79,
  },
  Object {
    "dividends": 790996.36,
    "salary": 180389,
    "surplus": 1480000,
    "total": 933615.36,
  },
  Object {
    "dividends": 796340.9299999999,
    "salary": 172721,
    "surplus": 1490000,
    "total": 932997.9299999999,
  },
  Object {
    "dividends": 801685.5,
    "salary": 165052,
    "surplus": 1500000,
    "total": 932324.5,
  },
  Object {
    "dividends": 807030.0700000001,
    "salary": 157384,
    "surplus": 1510000,
    "total": 931600.0700000001,
  },
  Object {
    "dividends": 812374.64,
    "salary": 149714,
    "surplus": 1520000,
    "total": 930872.64,
  },
  Object {
    "dividends": 817719.21,
    "salary": 142046,
    "surplus": 1530000,
    "total": 930148.21,
  },
  Object {
    "dividends": 823063.78,
    "salary": 134377,
    "surplus": 1540000,
    "total": 929422.78,
  },
  Object {
    "dividends": 828408.35,
    "salary": 126709,
    "surplus": 1550000,
    "total": 928698.35,
  },
  Object {
    "dividends": 833752.9199999999,
    "salary": 119040,
    "surplus": 1560000,
    "total": 927972.9199999999,
  },
  Object {
    "dividends": 839097.49,
    "salary": 111372,
    "surplus": 1570000,
    "total": 927247.49,
  },
  Object {
    "dividends": 844442.06,
    "salary": 103703,
    "surplus": 1580000,
    "total": 926522.06,
  },
  Object {
    "dividends": 849786.63,
    "salary": 96017,
    "surplus": 1590000,
    "total": 925784.63,
  },
  Object {
    "dividends": 855131.2,
    "salary": 88192,
    "surplus": 1600000,
    "total": 924935.2,
  },
  Object {
    "dividends": 860475.77,
    "salary": 80366,
    "surplus": 1610000,
    "total": 924085.77,
  },
  Object {
    "dividends": 865820.3400000001,
    "salary": 72541,
    "surplus": 1620000,
    "total": 923236.3400000001,
  },
  Object {
    "dividends": 871164.9099999999,
    "salary": 64716,
    "surplus": 1630000,
    "total": 922386.9099999999,
  },
  Object {
    "dividends": 876509.48,
    "salary": 56891,
    "surplus": 1640000,
    "total": 921538.48,
  },
  Object {
    "dividends": 881854.05,
    "salary": 49065,
    "surplus": 1650000,
    "total": 920689.05,
  },
  Object {
    "dividends": 887198.62,
    "salary": 41240,
    "surplus": 1660000,
    "total": 919839.62,
  },
  Object {
    "dividends": 892543.19,
    "salary": 33415,
    "surplus": 1670000,
    "total": 918991.19,
  },
  Object {
    "dividends": 897887.76,
    "salary": 25590,
    "surplus": 1680000,
    "total": 918142.76,
  },
  Object {
    "dividends": 903232.3300000001,
    "salary": 17764,
    "surplus": 1690000,
    "total": 917292.3300000001,
  },
  Object {
    "dividends": 908576.9,
    "salary": 9939,
    "surplus": 1700000,
    "total": 916443.9,
  },
]
`;
